flipperdevices/flipperzero-firmware
# fbt: Fix SCons error if PYTHONPATH set on Windows

# What's new

- On Windows, Python may fail to find the correct modules if `PYTHONPATH` is set in the environment. This PR ensures `PYTHONPATH` is unset when setting up fbtenv.

# Verification 

Before applying the patch:
- Install a local copy of Python that is a different version from the one included in the toolchain and set `PYTHONPATH` or `PY_HOME` (if present in `PYTHONPATH`) to the path of that installation
- Restart system, and run `fbt.cmd`. It should potentially error
  ```
  PS C:\projects\flipper_zero\flipperzero-firmware> .\fbt.cmd
  SCons import failed. Unable to find engine files in:
    C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\..
  Traceback (most recent call last):
    File "C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\scons.py", line 89, in <module>
      import SCons.Script
    File "C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\..\SCons\Script\__init__.py", line 71, in <module>
      import SCons.Action
    File "C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\..\SCons\Action.py", line 112, in <module>
      import SCons.Debug
    File "C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\..\SCons\Debug.py", line 105, in <module>
      from SCons.compat.win32 import get_peak_memory_usage
    File "C:\projects\flipper_zero\flipperzero-firmware\lib\scons\scripts\..\SCons\compat\win32.py", line 27, in <module>
      import ctypes
    File "C:\Users\cyanic\AppData\Local\Programs\Python\Python39\Lib\ctypes\__init__.py", line 8, in <module>
      from _ctypes import Union, Structure, Array
  ImportError: DLL load failed while importing _ctypes: %1 is not a valid Win32 application.
  ```

After applying the patch:
- Run `fbt.cmd` again. There should not be a Python error

# Checklist (For Reviewer)
